src/EnergyPlus/IntegratedHeatPump.cc

```diff
@@ -2451,10 +2451,11 @@ void DecideWorkMode(EnergyPlusData &state,
         state.dataLoopNodes->Node(state.dataIntegratedHP->IntegratedHeatPumps(DXCoilNum).WaterOutletNodeNum).Temp =
             state.dataLoopNodes->Node(state.dataIntegratedHP->IntegratedHeatPumps(DXCoilNum).WaterInletNodeNum).Temp;
 
-        int tankType = state.dataIntegratedHP->IntegratedHeatPumps(DXCoilNum).WHtankType;
+        DataPlant::PlantEquipmentType tankType = state.dataIntegratedHP->IntegratedHeatPumps(DXCoilNum).WHtankType;
 
-        if ((tankType == DataPlant::TypeOf_WtrHeaterMixed) || (tankType == DataPlant::TypeOf_WtrHeaterStratified) ||
-            (tankType == DataPlant::TypeOf_ChilledWaterTankMixed) || (tankType == DataPlant::TypeOf_ChilledWaterTankStratified)) {
+        if ((tankType == DataPlant::PlantEquipmentType::WtrHeaterMixed) || (tankType == DataPlant::PlantEquipmentType::WtrHeaterStratified) ||
```

Comment:
Consider switch/case or bitfields instead of comparisons to five different things.